prow/pjutil/tot.go

```diff
@@ -76,7 +76,7 @@ func PeriodicToJobSpec(periodic config.Periodic) *downwardapi.JobSpec {
 
 // GetBuildID calls out to `tot` in order
 // to vend build identifier for the job
-func GetBuildID(name, totURL string) (string, error) {
+func GetBuildID(name, totURL string, sleep time.Duration) (string, error) {
```

Comment:
This is being used like a boolean flag that indicates if we are testing or not. That smells to me.
Instead of adding an extra parameter that all 'real' callers will specify as `100 * time.Millisecond` and all test callers specify as `0`, we could mock one of the functions for testing.
Example:
https://github.com/kubernetes/test-infra/blob/fe0a9926c1c3d0a9d94e0d3c2f755dbdbc34d892/prow/external-plugins/needs-rebase/plugin/plugin.go#L43
https://github.com/kubernetes/test-infra/blob/fe0a9926c1c3d0a9d94e0d3c2f755dbdbc34d892/prow/external-plugins/needs-rebase/plugin/plugin_test.go#L140-L142
WDYT?

Reply:
good point, forgot you can do that in go :-)

Reply:
done